### ISSUE TITLE
[Bug] The battery's isSupported is broken.

### ISSUE THREAD

**Issue by Jesse205** [vueuse/vueuse#3688]
### Describe the bug

Using only `xxx in xxx` to determine support can be problematic.

![image](https://github.com/vueuse/vueuse/assets/51242302/1014f63d-b2e3-45bb-9ee5-0ad743dc62a9)

I am using a new version of Microsoft Edge and my computer is a battery free computer.

To fix this, just add `typeof navigator.getBattery === 'function'`

### Reproduction

https://stackblitz.com/edit/vitejs-vite-tuyecr?file=src%2FApp.vue

### System Info

```Shell
System:
    OS: Windows 10 10.0.19045
    CPU: (4) x64 Intel(R) Core(TM) i3-4170 CPU @ 3.70GHz
    Memory: 407.99 MB / 11.87 GB
  Binaries:
    Node: 21.2.0 - D:\Program Files\nodejs\node.EXE
    Yarn: 1.22.21 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 10.2.3 - D:\Program Files\nodejs\npm.CMD
    pnpm: 8.11.0 - ~\AppData\Roaming\npm\pnpm.CMD
  Browsers:
    Edge: Chromium (120.0.2210.77)
    Internet Explorer: 11.0.19041.3636
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by Jesse205** [comment]
Already exists pr https://github.com/vueuse/vueuse/pull/3634

**Comment by Jesse205** [comment]
I looked at other functions and it didn't added typeof either. Perhaps other functions also have this kind of bug?

**Comment by Doctor-wu** [comment]
> Already exists pr #3634

Hi @Jesse205 , That PR appears to be not active, welcome to commit a new one :)

**Comment by Doctor-wu** [comment]
> > Already exists pr #3634
> 
> Hi @Jesse205 , That PR appears to be not active, welcome to commit a new one :)

That author just replied, ignore this🥹

**Comment by Jesse205** [comment]
I found that `delete navigator.getBattery` can resets the getBattery function. And the problem is not reproduced in other browsers. I think it might be some extension I installed that makes `navigator.getBattery` `undefined` or only `http://xxx` cannot use `navigator.getBattery`.

Browers:
- Chrome: v120.0.6099.130, no extensions
- Cent Browser: v5.0.1002.354 (official release) (32-bit) (Chromium 102.0.5005.167), only Tampermonkey, no scripts
- Edge: v120.0.2210.91 (official version) (64-bit), very many extensions

Chrome(<https://cn.bing.com/chrome/newtab>):
![image](https://github.com/vueuse/vueuse/assets/51242302/fbd3d1ab-eb76-490e-9688-7ba10e959cb3)

Chrome(<http://192.168.56.1:5173/>):
![image](https://github.com/vueuse/vueuse/assets/51242302/a4532008-dc01-4f5e-8aa6-1876721feb01)

Cent Browser(<https://hao.centbrowser.cn/>):
![image](https://github.com/vueuse/vueuse/assets/51242302/68934735-4c89-4583-99e9-6940b6ad3b93)

Cent Browser(<http://192.168.56.1:5173/>):
![image](https://github.com/vueuse/vueuse/assets/51242302/24e07bdd-f11d-4976-a4f9-cdb1b44c89ae)

Edge(<http://192.168.56.1:5173/>):
![image](https://github.com/vueuse/vueuse/assets/51242302/271ef0b7-05f4-4a41-8202-5637ead9fc0e)
![image](https://github.com/vueuse/vueuse/assets/51242302/8d597c73-9b75-4c00-84eb-cdf9bae5df50)

Edge(<https://www.hao123.com/>):
![image](https://github.com/vueuse/vueuse/assets/51242302/359e18c9-5f05-4100-9b72-503dd1b8fd19)

**Comment by ilyaliao** [comment]
Closed via #3634
